packages/autorest.python/autorest/_utils.py

```diff
@@ -34,7 +34,11 @@ def replace_upper_characters(m) -> str:
 
         return prefix + match_str
 
-    return re.sub("[A-Z]+", replace_upper_characters, name)
+    return (
+        re.sub("[A-Z]+", replace_upper_characters, name)
+        .replace(" ", "_")
+        .replace("__", "_")
+    )
 
```

Comment:
There is no strict rule for example name in swagger like: [Lists all application gateways in a resource group](https://github.com/Azure/azure-rest-api-specs/blob/c71a66dab813061f1d09982c2748a09317fe0860/specification/network/resource-manager/Microsoft.Network/stable/2022-05-01/applicationGateway.json#L273)